Исправил недочеты в ДЗ №4

AndreyDemuh opened the pull request:
Выполнил домашку по одномерным массивам + дополнительную часть. Есть сложности в работе циклов при обработке массивов. в README.md более подробно описал трудности с ДЗ.
В дополнительной части, задание №1 не могу реализовать многократный повторный ввод размера массива. По коду только 1 раз переспрашивает. Много вариантов перебирал. Вроде бы понимание как заложить логику есть, когда начинаю реализовывать, весь код по швам идет. Буду думать дальше.

AndreyDemuh committed:
Выполнил домашку по одномерным массивам + дополнительная часть. Есть сложности в работе циклов при обрабоке массивов. в README.md более подробно описал трудности с ДЗ.

artempolozov commented on src/Lesson04/HW_04_Array.java:
```diff
@@ -0,0 +1,206 @@
+        System.out.println(Arrays.toString(arr1));
+        for (int i = 0; i < arr1.length; i++) {
+            if (arr1[i] == a)
+                b = true;
```
если нашел, то нужно выйти из цикла

artempolozov commented on src/Lesson04/HW_04_Array.java:
```diff
@@ -0,0 +1,206 @@
+        }
+        sizeNewArray = arr2.length - countSizeNewArray;
+        int[] newArray = new int[sizeNewArray];
+        int indexNewArray = 0;
```
лучше внести в инициализационный блок цикла

artempolozov commented on src/Lesson04/HW_04_Array.java:
```diff
@@ -0,0 +1,206 @@
+            System.out.print("������ �� ������ ������ �������, ��������� ����:");
+            scn.next();
+
+        }
```
нужно окружить циклом (лучше while) и спрашивать размер до тех пор, пока не будет введено корректное число

AndreyDemuh committed:
Выполнил домашку по одномерным массивам + дополнительная часть. Есть сложности в работе циклов при обрабоке массивов. в README.md более подробно описал трудности с ДЗ.
UPD. Исправил недочеты. Задание 1 из доп.части доставило немного проблем. Возникла сложность в реализации повторного использованния сканнера.

AndreyDemuh commented:
Не заметил апрув на ДЗ. Сделал корректировку кода и закинул второй коммит. Прошу прощения
